subprojects/core/src/main/java/org/gradle/caching/BuildCacheService.java

```diff
@@ -34,6 +34,16 @@
  *     All other failures will be considered fatal and cause the Gradle build to fail.
  *     Fatal failures could include failing to read or write cache entries due to file permissions, authentication or corruption errors.
  * </p>
+ * <p>
+ *     Every build cache implementation should define a {@link org.gradle.caching.configuration.BuildCache} configuration type.
+ * </p>
+ * <p>
+ *     A build cache implementation constructor <b>must</b> conform to these requirements:
+ *     <ul>
+ *         <li>Must accept an instance of the appropriate {@code BuildCache} configuration object as its first parameter</li>
+ *         <li>Must be annotated with {@link javax.inject.Inject}</li>
+ *     </ul>
+ * </p>
```

Comment:
You do, unfortunately.  Since we're using the `DependencyInjectingInstantiator`, you have to use `@Inject` or have a zero-arg constructor. 

See: https://github.com/gradle/gradle/blob/c8143c9bcf412a2d19053b8c926c7787c1dc8087/subprojects/core/src/main/java/org/gradle/api/internal/DependencyInjectingInstantiator.java#L132-L156